### ISSUE TITLE
[Feature Request] Upload to Imgur account?

### ISSUE THREAD

**Issue by ipkpjersi** [Francesco149/sharenix#16]
Hello,

I think it would be awesome if Sharenix offered a feature to upload photos to an imgur account so you can easily keep track of all your images and so you can easily delete them on the fly, or access any images at any time from anywhere.

Thanks.

**Comment by Francesco149** [comment]
good idea, I'll look into the imgur API to see how it works and let you know

**Comment by Francesco149** [comment]
so i took a look at the api and it's already possible with the current version of sharenix:

this is a temporary solution until there's proper oauth support built in

visit [this page](https://api.imgur.com/oauth2/authorize?client_id=b972ecca954f246&response_type=token)
and authorize the application. It will redirect you to the homepage of imgur,
but with a token in the URL

the URL will look like

```
https://imgur.com/#access_token=something&... (other stuff we don't care about)
```

you want the value of access_token ("something" in this example).

the config is the exact same as the anonymous imgur upload except that instead
of the Client-ID you have ```Bearer something``` where "something" is your
access_token.

```
        {
            "Name": "imgur.com (account)",
            "RequestType": "POST",
            "Headers": {
                "Authorization": "Bearer something"
            },
            "RequestURL": "https://api.imgur.com/3/image",
            "FileFormName": "image",
            "Arguments": {
                "type": "file"
            },
            "ResponseType": "Text",
            "URL": "$json:data.link$",
            "DeletionURL": "https://imgur.com/delete/$json:data.deletehash$"
        },
```

this works fine, however the access token will expire after about 1 month and
you will need to repeat the procedure to acquire a new one.

this will be fixed when proper OAuth handling is implemented.

**Comment by ipkpjersi** [comment]
I tried it out and it worked perfectly, thank you so much! I can't wait for proper OAuth support, hopefully I will only have to manually get the access token once before proper OAuth support gets implemented. ;)

**Comment by ipkpjersi** [comment]
So it's almost been about a month, so I figured I would ask - has any progress been made on this? I completely understand if not, I'm just curious. Thanks.

**Comment by Francesco149** [comment]
havent had time, also the main problem is every service might have a slightly different way of creating/refreshing/expiring the oauth tokens so maybe i should make it a plugin like twitter
